internal/pkg/deploy/cloudformation/env.go

```diff
@@ -77,6 +77,24 @@ func (cf CloudFormation) EnvironmentTemplate(appName, envName string) (string, e
 	return cf.cfnClient.TemplateBody(stackName)
 }
 
+// EnvironmentUsesLegacyServiceDiscovery returns true if the environment has been upgraded and services should continue using the legacy namespace.
+func (cf CloudFormation) EnvironmentUsesLegacySvcDiscovery(appName, envName string) (bool, error) {
+	stackName := stack.NameForEnv(appName, envName)
+	descr, err := cf.cfnClient.Describe(stackName)
+	if err != nil {
+		return false, err
+	}
+	stackOutputs := make(map[string]string)
+	for _, output := range descr.Outputs {
+		stackOutputs[*output.OutputKey] = *output.OutputValue
+	}
+	output, ok := stackOutputs[stack.EnvOutputLegacyServiceDiscovery]
+	if !ok || output != "false" {
+		return true, nil
+	}
```

Comment:
Do we need the `UseLegacyServiceDiscovery` output? Can we instead check if we have the new output `ServiceDiscoveryNamespaceIDWithEnv` in the template?